### PR TITLE
Created Note Listing Page

### LINKED ISSUE #2: Create note listing page UI

### PR DESCRIPTION
**Pull request by CalvinChiramal**
Fixes #2 
@yedhink Please review notes page.

### CODE REVIEW

**Comment by yedhink** [comment]
@Calvinwilson99 https://www.loom.com/share/83f9d2aa475a4adcb09a4b897733746f

File: app/javascript/src/components/Notes/Note.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+
+import { MenuVertical, Clock } from "neetoicons";
+import { Dropdown, Typography, Tag, Avatar, Tooltip } from "neetoui";
+import { getTimeFromNow, getDayAndTimeFromDate } from "utils";
+
+const Note = ({ title, description, createdAt, noteType }) => {
+  const { Menu, MenuItem } = Dropdown;
```
Comment (yedhink) [inline app/javascript/src/components/Notes/Note.jsx:8]: Keep this outside the component declaration.

File: app/javascript/src/components/Notes/Note.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <div className="mx-4 my-2 flex flex-col border border-solid p-5 shadow">
+      <div className="flex flex-row justify-between">
+        <Typography style="h4">{title}</Typography>
+        <Dropdown className="bg-transparent" icon={MenuVertical}>
+          <Menu>
+            <MenuItem.Button>Edit</MenuItem.Button>
+            <MenuItem.Button>Delete</MenuItem.Button>
```
Comment (yedhink) [inline app/javascript/src/components/Notes/Note.jsx:17]: Go through https://bigbinary.slack.com/archives/C01T5Q4DZAR/p1663060378971029 very carefully and fix the issue over here. Moving forward, I expect you to not repeat the same mistakes mentioned in that Slack thread.

Also go through https://bigbinary.slack.com/archives/C01T5Q4DZAR/p1663754719557349

File: app/javascript/src/components/Notes/index.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { Plus } from "neetoicons";
+import { Button } from "neetoui";
+import { Header } from "neetoui/layouts";
+
+import { NOTES } from "./constants";
+import NavPanel from "./NavPanel";
+import Note from "./Note";
+
+const Notes = () => {
+  const [searchValue, setSearchValue] = useState("");
```
Comment (yedhink) [inline app/javascript/src/components/Notes/index.jsx:12]: Let's call this `searchTerm`

File: app/javascript/src/components/routeConstants.js [hunk 4]
```diff
@@ -1,43 +1,21 @@
...
+//   },
+//   {
+//     path: CONTACTS_PATH,
+//     component: Settings,
+//   },
+//   {
+//     path: SETTINGS_PATH,
+//     component: Settings,
+//   },
+// ];
```
Comment (yedhink) [inline app/javascript/src/components/routeConstants.js:21]: ?

File: logfile [hunk 5]
```diff
@@ -0,0 +1,9 @@
+2022-11-23 20:13:15.652 IST [8248] LOG:  starting PostgreSQL 14.6 (Homebrew) on x86_64-apple-darwin22.1.0, compiled by Apple clang version 14.0.0 (clang-1400.0.29.202), 64-bit
```
Comment (yedhink) [inline logfile:1]: Why's this file committed? Add this to the gitignore.

**Review by yedhink — COMMENTED** [review]
@Calvinwilson99

**Comment by CalvinChiramal** [comment]
> @Calvinwilson99 https://www.loom.com/share/83f9d2aa475a4adcb09a4b897733746f

I understand Yedhin. The mass file deletion was supposed to be a part of cleaning up the code base PR. But I didn't go through the entire file structure before raising and merging that PR, so I added it as part of my next PR. I should have raised a second PR for cleaning code base and then proceeded to make my changes.

**Comment by CalvinChiramal** [comment]
I have also merged my note deletion page PR(#10 ). I will create a separate PR addressing the issues in #9 and propagated in #10 plus the review comments of #10.

File: app/javascript/src/components/Notes/Note.jsx (hunk 2 above)
Comment (CalvinChiramal) [reply app/javascript/src/components/Notes/Note.jsx:17]: I found two issues here based on the thread:
- Line 14: Don't override with custom colours using className, use component provided styles or neeto-ui custom CSS properties.
- Line 17: Provide a danger style to the Delete button.

Is my understanding correct?

File: app/javascript/src/components/routeConstants.js (hunk 4 above)
Comment (CalvinChiramal) [reply app/javascript/src/components/routeConstants.js:21]: I thought of using a constant routes file for declaring all the routes in App.jsx, but decided against it. I did not remove the unnecessary file. Will remove this file and improve my self review process.

File: app/javascript/src/components/Notes/Note.jsx (hunk 2 above)
Comment (yedhink) [reply app/javascript/src/components/Notes/Note.jsx:17]: Correct. 👍

**Pull request merged**